tasks/EchoesToFrameConverterTask.cpp

```diff
@@ -99,21 +100,27 @@ void EchoesToFrameConverterTask::addEchoesToFrame(Radar const& echo,
     if (angle < 0) {
         angle = 2 * M_PI + angle;
     }
-
-    int start_angle_unit = round(angle / echo.step_angle.getRad());
-    int angles_in_a_frame = round(2 * M_PI / echo.step_angle.getRad());
+    int start_angle_unit = round(angle / abs(echo.step_angle.getRad()));
+    int angles_in_a_frame = round(2 * M_PI / abs(echo.step_angle.getRad()));
+    int signal = abs(echo.step_angle.getRad()) / echo.step_angle.getRad();
     for (int current_angle = 0;
          current_angle < static_cast<int>(echo.sweep_timestamps.size());
          current_angle++) {
+
+        int echo_position =
+            (start_angle_unit + current_angle * signal) % angles_in_a_frame;
+        if (echo_position < 0) {
+            echo_position += angles_in_a_frame;
+        }
         std::copy(echo.sweep_data.begin() + current_angle * echo.sweep_length,
             echo.sweep_data.begin() + (current_angle + 1) * echo.sweep_length,
-            m_echoes.begin() + echo.sweep_length * ((start_angle_unit + current_angle) %
-                                                       angles_in_a_frame));
+            m_echoes.begin() + echo.sweep_length * echo_position);
     }
 }
 
 void EchoesToFrameConverterTask::publishFrame()
 {
+    m_cv_frame = 0;
```

Comment:
Why is it global?

Reply:
it is faster just to erase the cv matrix instead of creating a new one.